taar/recommenders/collaborative_recommender.py

```diff
@@ -92,19 +98,22 @@ def recommend(self, client_data, limit, extra_data={}):
             # We don't really need to show the items we requested. They will always
             # end up with the greatest score. Also filter out legacy addons from the
             # suggestions.
-            hashed_id = str(addon.get("id"))
-            if (hashed_id in installed_addons or
-                    hashed_id not in self.addon_mapping or
-                    self.addon_mapping[hashed_id].get("isWebextension", False) is False):
+            hashed_id = positive_hash(addon.get("id"))
+            str_hashed_id = str(hashed_id)
+            if (hashed_id in installed_addons_as_hashes or
+                    str_hashed_id not in self.addon_mapping or
+                    self.addon_mapping[str_hashed_id].get("isWebextension", False) is False):
                 continue
 
             dist = np.dot(user_factors_transposed, addon.get('features'))
             # Read the addon ids from the "addon_mapping" looking it
             # up by 'id' (which is an hashed value).
-            addon_id = self.addon_mapping[hashed_id].get("id")
+            addon_id = self.addon_mapping[str_hashed_id].get("id")
             distances[addon_id] = dist
 
         # Sort the suggested addons by their score and return the sorted list of addon
         # ids.
-        sorted_dists = sorted(distances.items(), key=op.itemgetter(1), reverse=True)
+        sorted_dists = sorted(distances.items(),
```

Comment:
nit: any reason this line was changed (e.g. PEP8 compliance or so)? If so, let's change it elsewhere too. Otherwise, let's drop this change and use the full available line length